boards/FSM/Test/Src/state_machine_test.cpp

```diff
@@ -0,0 +1,94 @@
+#include "fff.h"
+#include "gtest/gtest.h"
+
+extern "C"
+{
+#include "App_SharedStateMachine.h"
+#include "states/App_AirOpenState.h"
+#include "states/App_AirClosedState.h"
+
+    DEFINE_FFF_GLOBALS;
+    FAKE_VOID_FUNC(
+        send_non_periodic_msg_FSM_STARTUP,
+        struct CanMsgs_fsm_startup_t *);
+    FAKE_VOID_FUNC(
+        send_non_periodic_msg_FSM_AIR_SHUTDOWN,
+        struct CanMsgs_fsm_air_shutdown_t *);
+    FAKE_VOID_FUNC(
+        send_non_periodic_msg_FSM_WATCHDOG_TIMEOUT,
+        struct CanMsgs_fsm_watchdog_timeout_t *);
+}
+
+class FsmStateMachineTest : public testing::Test
+{
+  protected:
+    void SetUp() override
+    {
+        can_tx_interface = App_CanTx_Create(
+            send_non_periodic_msg_FSM_STARTUP,
+            send_non_periodic_msg_FSM_WATCHDOG_TIMEOUT,
+            send_non_periodic_msg_FSM_AIR_SHUTDOWN);
+        can_rx_interface = App_CanRx_Create();
+        world = App_FsmWorld_Create(can_tx_interface, can_rx_interface);
+
+        // Default to starting the state machine in the `AIR_OPEN` state
+        state_machine =
+            App_SharedStateMachine_Create(world, App_GetAirOpenState());
+
+        // Reset fake functions
+        RESET_FAKE(send_non_periodic_msg_FSM_STARTUP);
+        RESET_FAKE(send_non_periodic_msg_FSM_AIR_SHUTDOWN);
+        RESET_FAKE(send_non_periodic_msg_FSM_WATCHDOG_TIMEOUT);
+    }
+
+    void SetInitialState(const struct State *const initial_state)
```

Comment:
Might be worth it to make a `class SharedStateMachineTest` so we don't have to write `SetInitialState` multiple times?

i.e.
`class SharedStateMachineTest : public testing::Test`
`class FsmStateMachineTest : public SharedStateMachineTest`

Reply:
That seems like a reasonable idea to me, just keep in mind that it will be a bit of a hack in the same way that `App_SharedStateMachine` is because we have a different `World` for each board.

Reply:
Ugh yeah I definitely thought about that too. I think i will just leave it as-as for now. If more common functions like this show up in the future we can refactor then.